onnxruntime/core/framework/tensor.cc

```diff
@@ -60,8 +60,21 @@ void Tensor::InitOrtValue(MLDataType p_type, const TensorShape& shape, void* p_d
 }
 
 size_t Tensor::SizeInBytes() const {
+  int64_t size = shape_.Size();
+#ifdef ENABLE_TRAINING
+  if (!IsContiguous()) {
+    size = 1;
+    for (size_t dim = 0; dim < shape_.NumDimensions(); ++dim) {
+      if (shape_[dim] == 0) {
+        size = 0;
+        break;
+      }
+      size += strides_[dim] * (shape_[dim] - 1);
```

Comment:
The answer for this is same as for above comment. If we want to keep the real storage size for strided tensor, then we need to change the Tensor initialization code to pass the real storage size as well as the strides. But current ORT's stride tensor is not possible to own the storage, so this information is not useful for strided tensors. For any strided tensor, if the retuned size can cover all its data, it's enough. The original tensor (always contiguous for now) is keeping the actual allocation size and will take responsibility to free the storage.

Reply:
Thanks, this is functionally OK because the returned size is sufficient for a copy, but is imprecise. I will resolve the comment for this PR - we can discuss more over email.